### ISSUE TITLE
DiagnosticReport report in json format

### ISSUE THREAD

**Issue by elliottmatt** [NimaAra/Easy.Common#5]
I love the idea behind the DiagnosticReport but would it be possible to return in a JSON string rather than a "prettified" report? I love all the information but my logging system wouldn't take in that very well as a preformatted string.

Thanks.

**Comment by NimaAra** [comment]
I am not too keen on returning _JSON_ what I can do for you however, is return a rich _object_ instead of `string` whose `ToString()` returns the current _prettified_ report; You can then choose to serialize the object however you see fit.

How does that sound?

**Comment by elliottmatt** [comment]
I have a function somewhat similar to this and I return a string that is json inside it. See below. (I've censored/removed anything sensitive.) (I also return with the json inside not prettified, i just prettified it below so you can easily see how the sections/data can be nested).

```
{
	"CommandLine": "c:\\windows\\system32\\inetsrv\\w3wp.exe -morestuff",
	"CurrentDirectory": "c:\\windows\\system32\\inetsrv",
	"CurrentManagedThreadId": "63",
	"Is64BitOperatingSystem": "True",
	"Is64BitProcess": "True",
	"MachineName": "serverNameHere",
	"OSVersion": "Microsoft Windows",
	"ProcessorCount": "2",
	"UserDomainName": "DOMAINNAME",
	"UserName": "server_name",
	"Version": "4.0.30319.42000",
	"ServerCurrentTime": "1/1/2011 1:11:11 PM",
	"SystemUptime": "1.11:11:11.11111111",
	"ScriptTimeout": "00:11:11",
	"Trust Level": "Unrestricted",
	"EnvironmentVariables": {
		"ALLUSERSPROFILE": "C:\\ProgramData",
		"APP_POOL_ID": "app",
		"APPDATA": "C:\\Windows\\system32\\config\\systemprofile\\AppData\\Roaming",
		"CommonProgramFiles": "C:\\Program Files\\Common Files",
		"COMPUTERNAME": "serverNameHere",
		"ComSpec": "C:\\Windows\\system32\\cmd.exe",
		"FP_NO_HOST_CHECK": "NO",
		"NUMBER_OF_PROCESSORS": "8",
		"OS": "Windows_NT",
		"Path": "C:\\Windows\\system32;C:\\Windows;C:\\Windows\\System32\\Wbem",
		"PATHEXT": ".COM;.EXE;.BAT;.CMD;.VBS;.VBE;.JS;.JSE;.WSF;.WSH;.MSC",
		"PROCESSOR_ARCHITECTURE": "AMD64",
		"PROCESSOR_IDENTIFIER": "Intel64 Family 6 Model 45 Stepping 2, GenuineIntel",
		"PROCESSOR_LEVEL": "6",
		"PROCESSOR_REVISION": "2d02",
		"ProgramData": "C:\\ProgramData",
		"ProgramFiles": "C:\\Program Files",
		"ProgramFiles(x86)": "C:\\Program Files (x86)",
		"ProgramW6432": "C:\\Program Files",
		"PSModulePath": "C:\\Windows\\system32\\WindowsPowerShell\\v1.0\\Modules\\",
		"PUBLIC": "C:\\Users\\Public",
		"SystemDrive": "C:",
		"SystemRoot": "C:\\Windows",
		"TEMP": "C:\\Windows\\TEMP",
		"TMP": "C:\\Windows\\TEMP",
		"USERDOMAIN": "DOMAINNAME",
		"USERNAME": "username$",
		"USERPROFILE": "C:\\Windows\\system32\\config\\systemprofile",
		"windir": "C:\\Windows"
	},
	"ServerProcessor": {
		"Number of Processors": "8",
		"Processor Id": "Intel64 Family 6 Model 45 Stepping 2, GenuineIntel",
		"Processor Type": "255",
		"Processor Level": "8",
		"Processor OEM Id": "9",
		"Page Size": "4096"
	},
	"MemoryInfo": {
		"CurrentWorkingSet": "1,206.05 MB",
		"Physical Memory Size": "4,095.55 MB",
		"Physical Free Memory Size": "23 bytes",
		"PageFile Size": "3,480.06 MB",
		"Available PageFile Size": "2 bytes",
		"Virtual Memory Size": "4,095.55 MB",
		"Available Memory Size": "27 bytes",
		"Memory Load": "88.00 %"
	},
	"Session": {
		"SessionToken1": "SomeData"
	}
}
```

**Comment by NimaAra** [comment]
I understand but I do not want the library to concern itself with any serialization logic hence my suggestion for a rich object which you can serialize and get the similar format you have pasted.

**Comment by elliottmatt** [comment]
Perhaps that would work then? So your rich object would contain the different sections in them and I could enumerate through to format appropriately?

**Comment by NimaAra** [comment]
It will be something like:

```csharp
public sealed class Report
{
	public IDictionary<string, string> System { get; set; }
	public IDictionary<string, string> Process { get; set; }
	...	
}
```

**Comment by elliottmatt** [comment]
That would be perfect!

**Comment by NimaAra** [comment]
Okay, cool. Leave it with me and I will add it in the next release hopefully by next _Monday_.

**Comment by pldmgg** [comment]
Awesome! This is exactly what I came to the Issues section to see! Thanks again for creating this!

**Comment by NimaAra** [comment]
This is now available in: [v2.6.7](https://github.com/NimaAra/Easy.Common/releases/tag/Easy.Common-release-v2.6.7)
